rewrite-pattern.js

```diff
@@ -700,7 +700,7 @@ const rewritePattern = (pattern, flags, options) => {
 	config.transform.unicodeSetsFlag = config.flags.unicodeSets && transform(options, 'unicodeSetsFlag');
 
 	// unicodeFlag: 'transform' implies unicodePropertyEscapes: 'transform'
-	config.transform.unicodePropertyEscapes = config.flags.unicode && (
+	config.transform.unicodePropertyEscapes = (config.flags.unicode || config.flags.unicodeSets) && (
```

Comment:
Ah I see, there is a `transform(options, 'unicodePropertyEscapes')` check later so that `unicodePropertyEscapes: false` will still preserve the property escapes.